pandas/core/window.py

```diff
@@ -321,16 +321,21 @@ def validate(self):
         if isinstance(window, (list, tuple, np.ndarray)):
             pass
         elif com.is_integer(window):
-            try:
-                import scipy.signal as sig
-            except ImportError:
-                raise ImportError('Please install scipy to generate window '
-                                  'weight')
-
-            if not isinstance(self.win_type, compat.string_types):
-                raise ValueError('Invalid win_type {0}'.format(self.win_type))
-            if getattr(sig, self.win_type, None) is None:
-                raise ValueError('Invalid win_type {0}'.format(self.win_type))
+            if window < 0:
```

Comment:
Without this, it does behave buggy
>>> s = pd.Series(np.random.randn(5))
>>> s.rolling(-1, win_type='boxcar')
Window [window=-1,center=False,win_type=boxcar,axis=0]
>>> s.rolling(-1, win_type='boxcar').mean()
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "pandas/core/window.py", line 449, in mean
    return self._apply_window(mean=True, **kwargs)
  File "/home/anonymous/Desktop/pandas-priyankjain/pandas/core/window.py", line 389, in _apply_window
    window = self._prep_window(**kwargs)
  File "pandas/core/window.py", line 371, in _prep_window
    return sig.get_window(win_type, window).astype(float)
  File "pandas_dev/lib/python3.4/site-packages/scipy/signal/windows.py", line 1694, in get_window
    return winfunc(*params)
  File "pandas_dev/lib/python3.4/site-packages/scipy/signal/windows.py", line 59, in boxcar
    return np.ones(M, float)
  File "pandas_dev/lib/python3.4/site-packages/numpy/core/numeric.py", line 190, in ones
    a = empty(shape, dtype, order)
ValueError: negative dimensions are not allowed